update hierarchical BNN notebook with some nice figures

murphyk opened the pull request:
update hierarchical BNN notebook with some nice figures.
Code has not changed.

rlouf commented:
Thanks @canyon289 ! We agreed over email that I'd do the merge myself. Should have commented here.

canyon289 commented:
Ah sorry. Carry on you two :) I'll get out of the way

rlouf commented on notebooks/bnn_hierarchical_blackjax.ipynb:
```diff
@@ -35,7 +35,7 @@
       "source": [
         "# Hierarchical Bayesian neural networks\n",
         "\n",
```
This commit is not doing anything, and I don't see any difference with the notebook that is already on master. Are you sure you opened a PR with the right branch?

murphyk replied:
You are right, it was the wrong version. I've updated it.

rlouf reviewed (changes requested):
I can't find any difference between this notebook and the one on the `main` branch, maybe you didn't push the correct branch?

murphyk commented:
I fixed this and added the new notebook, but get this commit error

Run ***@***.***
nstall pre-commit
Cache Size: ~81 MB (85222484 B)
/usr/bin/tar --use-compress-program zstd -d -xf
/home/runner/work/_temp/9370fc14-9e34-4f63-bfeb-61c1a69d6231/cache.tzst -P
-C /home/runner/work/blackjax/blackjax
/opt/hostedtoolcache/Python/3.9.7/x64/bin/pre-commit run
--show-diff-on-failure --color=always --all-files
Check for merge
conflicts................................................Passed
Check Toml...........................................(no files to
check)Skipped
Check
Yaml...............................................................Passed
Debug Statements
(Python)................................................Passed
Fix End of
Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing notebooks/HierarchicalBNN.ipynb

Don't commit to
branch..................................................Skipped
Fix
requirements.txt.....................................................Passed
Trim Trailing
Whitespace.................................................Passed
isort....................................................................Passed
pyupgrade................................................................Passed
flake8...................................................................Passed
black....................................................................Passed
mypy.....................................................................Passed
nbqa-black...............................................................Failed
- hook id: nbqa-black
- files were modified by this hook

reformatted notebooks/HierarchicalBNN.ipynb
All done! ✨ 🍰 ✨
1 file reformatted, 5 files left unchanged.

nbqa-pyupgrade...........................................................Failed
- hook id: nbqa-pyupgrade
- exit code: 1
- files were modified by this hook

Rewriting notebooks/HierarchicalBNN.ipynb

nbqa-isort...............................................................Failed
- hook id: nbqa-isort
- files were modified by this hook

Fixing /home/runner/work/blackjax/blackjax/notebooks/HierarchicalBNN.ipynb

nbqa-flake8..............................................................Failed
- hook id: nbqa-flake8
- exit code: 1

notebooks/HierarchicalBNN.ipynb:cell_2:11:1: F401 'sklearn' imported but
unused
notebooks/HierarchicalBNN.ipynb:cell_2:13:1: F401 'jax.jit' imported but
unused
notebooks/HierarchicalBNN.ipynb:cell_2:14:1: F401 'jax.random.normal'
imported but unused
notebooks/HierarchicalBNN.ipynb:cell_2:15:1: F401 'sklearn.datasets'
imported but unused
notebooks/HierarchicalBNN.ipynb:cell_2:17:1: F401
'sklearn.model_selection.train_test_split' imported but unused
notebooks/HierarchicalBNN.ipynb:cell_22:9:9: F841 local variable 'contour'
is assigned to but never used
notebooks/HierarchicalBNN.ipynb:cell_26:3:5: F841 local variable
'output_dim' is assigned to but never used
notebooks/HierarchicalBNN.ipynb:cell_31:9:9: F841 local variable 'contour'
is assigned to but never used

pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit
run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/notebooks/HierarchicalBNN.ipynb
b/notebooks/HierarchicalBNN.ipynb
index 9924d91..bd91796 100644
--- a/notebooks/HierarchicalBNN.ipynb
+++ b/notebooks/HierarchicalBNN.ipynb
@@ -1,1001 +1,1068 @@
 {

On Wed, Sep 29, 2021 at 11:13 AM Rémi Louf ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> I can't find any difference between this notebook and the one on the main
> branch, maybe you didn't push the correct branch?
> ------------------------------
>
> In notebooks/bnn_hierarchical_blackjax.ipynb
> <https://github.com/blackjax-devs/blackjax/pull/122#discussion_r718769388>
> :
>
> > @@ -35,7 +35,7 @@
>        "source": [
>          "# Hierarchical Bayesian neural networks\n",
>          "\n",
>
> This commit is not doing anything, and I don't see any difference with the
> notebook that is already on master. Are you sure you opened a PR with the
> right branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/blackjax-devs/blackjax/pull/122#pullrequestreview-767035072>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDK6EF4VXC55R5M3TI3EIDUENJMXANCNFSM5EYEXHMQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

rlouf commented:
`pre-commit` runs a notebook linter which found issue with the code. I fixed these when I merged your previous PR, I'll do the same here.

Versioning notebooks is a bit of a pain. I'll see if we can keep example notebooks as markdown files in the repo using `jupytext` and render them with figures in the documentation.

rlouf commented:
I am merging and will deal with the errors in a new branch. Thank you so much for adding these figures!
